x-pack/plugins/alerting/server/application/rule/methods/bulk_enable/bulk_enable_rules.ts

```diff
@@ -281,10 +276,14 @@ const bulkEnableRulesWithOCC = async (
                   alertInstances: {},
                 },
                 scope: ['alerting'],
-                enabled: false, // we create the task as disabled, taskManager.bulkEnable will enable them by randomising their schedule datetime
```

Comment:
> This is a little worrisome! Perhaps. Briefly looking through the code path, it doesn't look like we do the schedule randomization anymore, since we skip the enable. Looks like that processing was done here:

Hmm.. good point. Do you think I can still keep the `enabled:false` for `tasksToSchedule` and let task manager's bulk enable take care of randomization? Or it could create an issue with our 2 steps SO update logic?

> Do we have a test for this? That some number of rules, created disabled, then enabled, all have "randomized" task runAt dates?

I did not see any test for randomized task runAt dates. I will add one.

Reply:
Ya, we talked about this some more in our E+C meeting this morning; we feel like we want that randomization. 

Seems like we could change the ` bulkEnable(taskIds: string[], runSoon: boolean = true)` function to modify the filter a bit:

https://github.com/elastic/kibana/blob/7fa010e50712f89e38f141badc5cf20fd12f007a/x-pack/plugins/task_manager/server/task_scheduling.ts#L155-L161

change 

```ts
filter: (task) => !!task.enabled,
```

to something like:

```ts
filter: (task) => !!task.enabled || !task.runAt,
```

Basically, test a field in the task to see if the task hasn't been run yet.  Thoughts, @mikecote ?

Reply:
Updated and added unit test in https://github.com/elastic/kibana/pull/189041/commits/3d37696ba1c7fe3d4867adfb116cdf26d6090307

Reply:
I'm not sure what adding `!task.runAt` helps with, but it feels it would be good to let task manager randomize all the tasks. Is there a concern leaving `enabled: false` in the code?

Reply:
hmm.. I don't see any concern with keeping `enabled: false`, but I have a limited knowledge about this scenario here 😄

Reply:
updated in https://github.com/elastic/kibana/pull/189041/commits/00bca22012273ce014cf2faa87536b9527768fb4